x/protocolpool/types/keys.go

```diff
@@ -15,7 +15,7 @@ const (
 	// GovModuleName duplicates the gov module's name to avoid a cyclic dependency with x/gov.
 	// It should be synced with the gov module's name if it is ever changed.
 	// See: https://github.com/cosmos/cosmos-sdk/blob/b62a28aac041829da5ded4aeacfcd7a42873d1c8/x/gov/types/keys.go#L9
-	GovModuleName = "gov"
+	GovModuleName = "gov" // TODO this should be distribution
```

Comment:
Why should it be distribution? Isn't this used as default authority for gov proposal messages?

Reply:
only distribution should be calling protocol pool as that is where the funds are coming from. If distribution authority becomes a multisig then users also control how much is sent to the protocol module, but we should always set the caller of another module to be the caller, not a sudo address imo.

Reply:
Okay, I think we talk about two different things. Your point makes sense. However this constant is used as the module authority for executing some messages. For those cases, we want gov to be the authority.

Reply:
ah okay, then the todo is more to add distribution key here for checking